foreman/data_refinery_foreman/surveyor/external_source.py

```diff
@@ -72,8 +43,24 @@ def determine_pipeline(self,
         """
         return
 
+    def downloader_task(self):
+        """Returns the downloader task for the source.
+
+        Returns the Celery Downloader Task name from the
+        data_refinery_workers project which should be queued to
+        download Batches discovered by this surveyor.
+        """
+        return DOWNLOADER_TASK_LOOKUP[self.source_type()]
+
     def handle_batches(self, batches: List[Batch]):
+        new_batches = []
         for batch in batches:
+            if not Batch.is_new_batch(batch):
```

Comment:
Hmmm - I'm not sure how much is being added by this `is_new_batch` function. Could we just call here (and wherever else this function is called) something like the original `if  Batch.objects.filter(name=batch.name).count() > 0:` line that the `is_new_batch` function calls (albeit somewhat differently)?